lib/doctest.js

```diff
@@ -378,7 +372,11 @@ function rewrite$js(options, input) {
   //     produced by step 6 (substituting "step 6" for "step 2").
 
   function getComments(input) {
-    return (esprima.parse (input, {comment: true, loc: true})).comments;
+    return (esprima.parse (input, {
+      comment: true,
+      loc: true,
+      sourceType: options.sourceType || 'script'
```

Comment:
Maybe better to provide it explicitly on every callsite.

Reply:
I agree. `||` makes me nervous.